contracts/staking/PoSValidatorManager.sol

```diff
@@ -17,10 +20,23 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
     // solhint-disable private-vars-leading-underscore
     /// @custom:storage-location erc7201:avalanche-icm.storage.PoSValidatorManager
     struct PoSValidatorManagerStorage {
+        /// @notice The minimum amount of stake required to be a validator.
         uint256 _minimumStakeAmount;
+        /// @notice The maximum amount of stake allowed to be a validator.
         uint256 _maximumStakeAmount;
+        /// @notice The minimum amount of time a validator must be staked for.
         uint64 _minimumStakeDuration;
+        /// @notice The reward calculator for this validator manager.
         IRewardCalculator _rewardCalculator;
+        /// @notice Maps the validationID to a mapping of delegator address to delegator information.
+        mapping(bytes32 validationID => mapping(address delegator => Delegator)) _delegatorStakes;
```

Comment:
I think it may make more sense here to create a notion of a `delegationID` (similar to `validationID`) to key these maps on.

Theoretically, an address should be able to delegate to the same validation ID multiple times.

The `delegationID` could be a hash of the (`validationID`,`nonce`), where the nonce is the nonce used in the `SetSubnetValidatorWeight` message that added the delegator.

Reply:
This idea came up [previously](https://github.com/ava-labs/teleporter/pull/504#discussion_r1735220238). I agree that using a unique delegation ID rather than the address is the way to go, but going to defer that to a followup.

Using the nonce rather than a timestamp (as originally suggested in the linked conversation) seems like the way to go.

Reply:
i like the (validationID, nonce) hash combo